src/exporter/ceph_exporter.cc

```diff
@@ -1,16 +1,31 @@
 #include "common/ceph_argparse.h"
 #include "common/config.h"
-#include "exporter/DaemonMetricCollector.h"
-#include "exporter/web_server.h"
+#include "common/debug.h"
 #include "global/global_init.h"
 #include "global/global_context.h"
-
+#include "global/signal_handler.h"
+#include "exporter/DaemonMetricCollector.h"
+#include "exporter/web_server.h"
 #include <boost/thread/thread.hpp>
 #include <iostream>
 #include <map>
 #include <string>
+#include <atomic>
+#include <chrono>
+#include <thread>
 
 #define dout_context g_ceph_context
+#define dout_subsys ceph_subsys_ceph_exporter
+
+static std::atomic<bool> stop_signal_received(false);
+
+static void handle_signal(int signum)
+{
+  ceph_assert(signum == SIGINT || signum == SIGTERM);
```

Comment:
should we handle SIGHUP here as well?

Reply:
No. SIGHUP is managed directly in the signals library